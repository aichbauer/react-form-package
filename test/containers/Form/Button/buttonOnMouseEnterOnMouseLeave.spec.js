import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Button,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form (validate) with disabled button | on mouse enter button', () => {
  const onMouseEnterMock = jest.fn();
  const onMouseLeaveMock = jest.fn();

  const myComponent = (
    <Form
      validate
    >
      <div>
        <Button
          id="submit"
          type="submit"
          disabled
          onMouseEnter={onMouseEnterMock}
          onMouseLeave={onMouseLeaveMock}
        >
          submit
        </Button>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);

  wrapper.find('button').simulate('mouseenter');
  wrapper.update();
  wrapper.find('button').simulate('mouseleave');
  wrapper.update();

  expect(onMouseEnterMock).toHaveBeenCalledTimes(1);
  expect(onMouseLeaveMock).toHaveBeenCalledTimes(1);
  expect(tree).toMatchSnapshot();
});
