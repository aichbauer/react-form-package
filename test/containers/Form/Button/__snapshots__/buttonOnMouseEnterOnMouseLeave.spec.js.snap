// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form (validate) with disabled button | on mouse enter button 1`] = `
<form>
  <div>
    <button
      className="rfp-button"
      disabled={false}
      id="submit"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      type="submit"
    >
      submit
    </button>
  </div>
</form>
`;
